Sources/SwiftProtobuf/DelimitedMessageUtil.swift

```diff
@@ -0,0 +1,215 @@
+// Sources/SwiftProtobuf/DelimitedMessageUtil.swift - Delimited support
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/master/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// Helpers to read/write message with a length prefix.
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+/// Additional errors for delimited message handing.
+public enum DelimitedMessageError: Error {
```

Comment:
I'm in favor of nesting; it would only add a couple characters to the catch sites and many users might just do a catch-all or `try?` anyway and it won't affect them.